mailpoet/lib/Newsletter/Sending/ScheduledTaskSubscribersRepository.php

```diff
@@ -125,6 +125,17 @@ public function deleteByScheduledTask(ScheduledTaskEntity $scheduledTask): void
       ->execute();
   }
 
+  public function setSubscribers(ScheduledTaskEntity $task, array $subscriberIds): void {
+    $this->deleteByScheduledTask($task);
```

Comment:
Do you expect the delete to be mostly a noop or can it sometimes delete many records that would be then recreated again below? (In which case, it might be better to delete after with `WHERE id NOT IN (...)`.)

Reply:
That is a good question and something that I haven't considered. Here, I basically implemented the same behavior that we had in the Paris code: https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/Models/ScheduledTaskSubscriber.php#L44

So I guess we can keep it as need and improve in the future if needed.